packages/components/src/components/hds/advanced-table/index.ts

```diff
@@ -0,0 +1,346 @@
+  @tracked private _isSelectAllCheckboxSelected?: boolean = undefined;
+
+  private _selectableRows: HdsAdvancedTableSelectableRow[] = [];
+  private _captionId = 'caption-' + guidFor(this);
+
+  get getSortCriteria(): string | HdsAdvancedTableSortingFunction<unknown> {
+    // get the current column
+    const currentColumn = this.args?.columns?.find(
+      (column) => column.key === this._sortBy
+    );
+    if (
+      // check if there is a custom sorting function associated with the current `sortBy` column (we assume the column has `isSortable`)
+      currentColumn?.sortingFunction &&
+      typeof currentColumn.sortingFunction === 'function'
+    ) {
+      return currentColumn.sortingFunction;
+    } else {
+      // otherwise fallback to the default format "sortBy:sortOrder"
+      return `${this._sortBy}:${this._sortOrder}`;
+    }
+  }
+
+  get columnWidths(): string[] | undefined {
+    const { columns } = this.args;
+    const widths: string[] = new Array(columns.length);
+    let hasCustomColumnWidth = false;
+
+    for (let i = 0; i < columns.length; i++) {
+      const column = columns[i];
+
+      if (column?.['width']) {
+        widths[i] = column.width;
+        if (!hasCustomColumnWidth) hasCustomColumnWidth = true;
+      }
+    }
+
+    return hasCustomColumnWidth ? widths : undefined;
+  }
+
+  get identityKey(): string | undefined {
+    // we have to provide a way for the consumer to pass undefined because Ember tries to interpret undefined as missing an arg and therefore falls back to the default
+    if (this.args.identityKey === 'none') {
+      return undefined;
+    } else {
+      return this.args.identityKey ?? '@identity';
+    }
+  }
+
+  get sortedMessageText(): string {
+    if (this.args.sortedMessageText) {
+      return this.args.sortedMessageText;
+    } else if (this._sortBy && this._sortOrder) {
+      // we should allow the user to define a custom value here (e.g., for i18n) - tracked with HDS-965
+      return `Sorted by ${this._sortBy} ${this._sortOrder}ending`;
+    } else {
+      return '';
+    }
+  }
+
+  get isSelectable(): boolean {
+    const { isSelectable = false } = this.args;
+
+    return isSelectable;
+  }
+
+  get isStriped(): boolean {
+    const { isStriped = false } = this.args;
+
+    return isStriped;
+  }
+
+  get density(): HdsAdvancedTableDensities {
+    const { density = DEFAULT_DENSITY } = this.args;
+
+    assert(
+      `@density for "Hds::Table" must be one of the following: ${DENSITIES.join(
+        ', '
+      )}; received: ${density}`,
+      DENSITIES.includes(density)
+    );
+
+    return density;
+  }
+
+  get valign(): HdsAdvancedTableVerticalAlignment {
+    const { valign = DEFAULT_VALIGN } = this.args;
+
+    assert(
+      `@valign for "Hds::Table" must be one of the following: ${VALIGNMENTS.join(
+        ', '
+      )}; received: ${valign}`,
+      VALIGNMENTS.includes(valign)
+    );
+
+    return valign;
+  }
+
+  get gridTemplateColumns(): string {
```

Comment:
[suggestion] - what do you think about adding a comment that expands a little on what kind of string this returns?